server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTreeBelowSGCachedImpl.java

```diff
@@ -331,6 +335,62 @@ public List<IMeasurementMNode> createAlignedTimeseries(
     }
   }
 
+  @Override
+  public Map<Integer, MetadataException> checkMeasurementExistence(
+      PartialPath devicePath, List<String> measurementList, List<String> aliasList) {
+    IMNode device = null;
+    try {
+      device = getNodeByPath(devicePath);
+    } catch (MetadataException e) {
+      return Collections.emptyMap();
+    }
+    try {
+      if (!device.isEntity()) {
+        return Collections.emptyMap();
+      }
+      Map<Integer, MetadataException> failingMeasurementMap = new HashMap<>();
+      for (int i = 0; i < measurementList.size(); i++) {
+        try {
+          if (store.hasChild(device, measurementList.get(i))) {
+            IMNode node = store.getChild(device, measurementList.get(i));
```

Comment:
Use ```node = store.getChild``` and judge ```node == null``` to avoid reading disk twice.

Reply:
done